core/server/master/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -140,6 +140,26 @@ FileInfo getFileInfo(AlluxioURI path, GetStatusContext context)
       throws AccessControlException, FileDoesNotExistException, InvalidPathException,
       UnavailableException, IOException;
 
+  /**
+   * Enumerates given path to given batch tracker.
+   * If the given path is a file, the list only
+   * contains a single object. If it is a directory, the resulting list contains all direct children
+   * of the directory.
+   * <p>
+   * This operation requires users to have READ permission on the path, and also
+   * EXECUTE permission on the path if it is a directory.
+   *
+   * @param path the path to get the {@link FileInfo} list for
+   * @param context the method context
+   * @param batchTracker the tracker that will receive batches of statuses
+   * @throws AccessControlException if permission checking fails
+   * @throws FileDoesNotExistException if the file does not exist
+   * @throws InvalidPathException if the path is invalid
+   */
+  void listStatus(AlluxioURI path, ListStatusContext context,
+      BatchTracker<ListStatusBatch> batchTracker) throws AccessControlException,
```

Comment:
Why do we need an intermediate class like `ListStatusBatch` and not just use `List<FileInfo>` directly?

Reply:
Otherwise I'd need to pass both `BatchTracker<List<FileInfo>>` and an `Integer` for denoting the batch size. `ListStatusBatch` combines the data and the control.